apps/qubit/modules/search/actions/advancedAction.class.php

```diff
@@ -450,6 +462,65 @@ static function addToQueryBool(&$queryBool, $operator, $queryField)
       }
   }
 
+  static function getDateRangeQuery($start, $end)
+  {
+    if (empty($start) && empty($end))
+    {
+      return;
+    }
+
+    $query = new \Elastica\Query\Bool();
+    $range = array();
+
+    if (!empty($start))
```

Comment:
Could we turn these 3 ifs into an if/else if/else? like:

if (!empty($start) && empty($end)) { ... } else if (!empty($end) && empty($start)) { ... } else { ...}

That way all those variables aren't assigned when I think they don't need to. I could be wrong with understanding what you're trying to do though.